homeassistant/components/lektrico/sensor.py

```diff
@@ -0,0 +1,314 @@
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        value=lambda data: float(data.power_l1) / 1000,
+    ),
+    LektricoSensorEntityDescription(
+        key="power_l2",
+        name="Power L2",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.POWER,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        value=lambda data: float(data.power_l2) / 1000,
+    ),
+    LektricoSensorEntityDescription(
+        key="power_l3",
+        name="Power L3",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.POWER,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        value=lambda data: float(data.power_l3) / 1000,
+    ),
+    LektricoSensorEntityDescription(
+        key="pf_l1",
+        name="PF L1",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.POWER_FACTOR,
+        native_unit_of_measurement=PERCENTAGE,
+        value=lambda data: float(data.power_factor_l1) * 100,
+    ),
+    LektricoSensorEntityDescription(
+        key="pf_l2",
+        name="PF L2",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.POWER_FACTOR,
+        native_unit_of_measurement=PERCENTAGE,
+        value=lambda data: float(data.power_factor_l2) * 100,
+    ),
+    LektricoSensorEntityDescription(
+        key="pf_l3",
+        name="PF L3",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.POWER_FACTOR,
+        native_unit_of_measurement=PERCENTAGE,
+        value=lambda data: float(data.power_factor_l3) * 100,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lektrico charger based on a config entry."""
+    coordinator: LektricoDeviceDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    _sensors_to_be_used: tuple[LektricoSensorEntityDescription, ...]
+    if coordinator.device_type == Device.TYPE_1P7K:
+        _sensors_to_be_used = SENSORS_FOR_CHARGERS + SENSORS_FOR_1_PHASE
+    elif coordinator.device_type == Device.TYPE_3P22K:
+        _sensors_to_be_used = SENSORS_FOR_CHARGERS + SENSORS_FOR_3_PHASE
+    elif coordinator.device_type == Device.TYPE_EM:
+        _sensors_to_be_used = (
+            SENSORS_FOR_LB_DEVICES + SENSORS_FOR_1_PHASE + SENSORS_FOR_LB_1_PHASE
+        )
+    elif coordinator.device_type == Device.TYPE_3EM:
+        _sensors_to_be_used = (
+            SENSORS_FOR_LB_DEVICES + SENSORS_FOR_3_PHASE + SENSORS_FOR_LB_3_PHASE
+        )
+    else:
+        return
+
+    async_add_entities(
+        LektricoSensor(
+            description,
+            coordinator,
+            entry.data[CONF_FRIENDLY_NAME],
+        )
+        for description in _sensors_to_be_used
+    )
+
+
+class LektricoSensor(CoordinatorEntity, SensorEntity):
```

Comment:
CoordinatorEntity[LektricoDeviceDataUpdateCoordinator]

Reply:
Thank you, I made the proposed change.